vllm/model_executor/layers/quantization/gptq.py

```diff
@@ -179,37 +181,40 @@ def create_weights(
             "input_dim": scale_and_zero_input_dim,
             "output_dim": 1,
         })
-        return {
-            "qweight": qweight,
-            "g_idx": g_idx,
-            "qzeros": qzeros,
-            "scales": scales,
-            "exllama_state": exllama_state,
-        }
+
+        layer.register_parameter("qweight", qweight)
+        set_weight_attrs(qweight, extra_weight_attrs)
+        layer.register_parameter("g_idx", g_idx)
+        set_weight_attrs(g_idx, extra_weight_attrs)
+        layer.register_parameter("qzeros", qzeros)
+        set_weight_attrs(qzeros, extra_weight_attrs)
+        layer.register_parameter("scales", scales)
+        set_weight_attrs(scales, extra_weight_attrs)
+
+        layer.exllama_state = exllama_state
```

Comment:
The `LinearLayer` does not need to know this info, so I think it should be `GPTQLinearMethod.exllama_state` ... It probably should not have been in the `weights` dict before

This will avoid having this dangling member

Reply:
ah actually I don't think this is possible since GPTQLinearMethod is a singleton across all layers

Reply:
Sounds good